bundles/org.openhab.binding.philipsair/src/main/java/org/openhab/binding/philipsair/internal/PhilipsAirHandler.java

```diff
@@ -0,0 +1,342 @@
+                logger.warn("Update channel {}: Unsupported value type {}", channelUID,
+                        value.getClass().getSimpleName());
+            }
+
+            updateState(channelUID, state);
+        }
+    }
+
+    @Override
+    protected void updateState(ChannelUID channelUID, State state) {
+        super.updateState(channelUID, state);
+    }
+
+    public @Nullable Object getValue(String channelId, @Nullable PhilipsAirPurifierData data,
+            @Nullable PhilipsAirPurifierDevice deviceInfo, @Nullable PhilipsAirPurifierFilters filters)
+            throws Exception {
+        String[] fields = StringUtils.split(channelId, ":");
+        String field = fields[fields.length - 1];
+        if (field.contains("#")) {
+            field = field.split("#")[1];
+        }
+
+        if (data != null) {
+            switch (field) {
+                case LED_LIGHT_BRIGHTNESS:
+                    return data.getLightLevel();
+                case DISPLAYED_INDEX:
+                    return Integer.toString(data.getDisplayIndex());
+                case BUTTONS_LIGHT:
+                    return data.getButtons() == 0 ? OnOffType.OFF : OnOffType.ON;
+                case POWER:
+                    return data.getPower() == 0 ? OnOffType.OFF : OnOffType.ON;
+                case PM25:
+                    return new QuantityType<Density>(data.getPm25(), DENSITY_UNIT);
+                case FAN_MODE:
+                    return data.getFanSpeed();
+                case CHILD_LOCK:
+                    return data.getChildLock() ? OnOffType.ON : OnOffType.OFF;
+                case AUTO_TIMEOFF:
+                    return data.getTimer();
+                case TIMER_COUNTDOWN:
+                    return data.getTimerLeft();
+                case MODE:
+                    return data.getMode();
+                case ALLERGEN_INDEX:
+                    return data.getAllergenLevel();
+                case AIR_QUALITY_NOTIFICATION_THRESHOLD:
+                    return data.getAqit();
+                case ERROR_CODE:
+                    return String.valueOf(data.getErrorCode());
+                case HUMIDITY:
+                    return new QuantityType<Dimensionless>(
+                            data.getHumidity() + getAirPurifierConfig().getHumidityOffset(), HUMIDITY_UNIT);
+                case HUMIDITY_SETPOINT:
+                    return data.getHumiditySetpoint();
+                case TEMPERATURE:
+                    return new QuantityType<>(data.getTemperature() + getAirPurifierConfig().getTemperatureOffset(),
+                            TEMPERATURE_UNIT);
+                case FUNCTION:
+                    return data.getFunction();
+                case WATER_LEVEL:
+                    return data.getWaterLevel();
+            }
+
+            if (deviceInfo != null) {
+                switch (field) {
+                    case SOFTWARE_VERSION:
+                        deviceInfo.getSoftwareVersion();

```

Comment:
I assume this was missing?
```suggestion
                        return deviceInfo.getSoftwareVersion();

```